impl/src/main/java/com/sun/faces/context/ExternalContextImpl.java

```diff
@@ -130,16 +107,11 @@ public ExternalContextImpl(ServletContext sc, ServletRequest request, ServletRes
 
         boolean enabled = ContextParamUtils.getValue(servletContext, SendPoweredByHeader, Boolean.class);
         if (enabled) {
-            ((HttpServletResponse) response).addHeader("X-Powered-By", "Faces/3.0");
+            ((HttpServletResponse) response).addHeader("X-Powered-By", "Faces/4.0");
```

Comment:
Nice catch! This is actually a Mojarra bug which needs to go into 4.0 branch via a separate bugticket.